### ISSUE TITLE
Documentation / Maturity?

### ISSUE THREAD

**Issue by Fohdeesha** [benjamin9999/pyperc#1]
Hi, this looks like a great project. Is it mature enough to use in production? I see the last commit was a few years ago. Is there documentation anywhere for actually setting up and configuring/using it and/or a list of features?

Thanks!

**Comment by usrlocalben** [comment]
Thanks, but I'm afraid it has just sat here idle for a while.

tl;dr - It could use some love.

Previously, I managed Linux servers that ran on MegaRAID boards.  Now I have vSphere/ESXi boxes instead, and MegaCLI runs inside the ESXi host.  I would like to come up with a way to get this running again with ESX, but it remains on the to-do list.

Most of the effort so far is accumulating MegaCLI output samples from different boards, disk-configs and raid-states to improve the parser.  If you were to try and get the code running, more than likely you will encounter new status codes and other constants that will need to be added to the data tables.

I have yet to start any API for operations that would modify the state, so there isn't much that can go wrong.  IIRC the only state-changing that occurs is related to polling the event logs.  (Which is also frustrating because in some configurations, the act of _getting_ the status from MegaCLI will result in more log output, so this may create a feedback loop)

**Comment by davidrenne** [comment]
Ben is a great dev @Fohdeesha.  He'll help out on any of your needs to improve his product.  I had the privilege of working with him in some shitty cubicles cranking out excellent solutions for a couple of years.
